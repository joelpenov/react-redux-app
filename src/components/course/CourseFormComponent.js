import React from "react";
import PropTypes from "prop-types";
import InputComponent from "../htmlElements/InputComponent";
import SelectComponent from "../htmlElements/SelectComponent";

function CourseFormComponent({ course, authors }) {
  const onChangeCourseName = e => {};

  const nameInputProps = {
    name: "courseName",
    label: "Course Name",
    onChange: onChangeCourseName,
    placeholder: "Course Name",
    value: "",
    error: ""
  };

  const categoryProps = {
    name: "courseCategory",
    label: "Category",
    onChange: onChangeCourseName,
    placeholder: "Category",
    value: "",
    error: "",
    inputType: "text"
  };

  let mappedAuthors = authors.map(a => {
    return { value: a.id, text: a.name };
  });

  const selectProps = {
    options: mappedAuthors,
    onChange: () => {},
    label: "Author",
    name: "Author",
    value: 0,
    error: ""
  };

  return (
    <div>
      <h1>{course.id ? "Edit " : "Add "} course</h1>
      <form>
        <InputComponent {...nameInputProps} />
        <SelectComponent {...selectProps} />
        <InputComponent {...categoryProps} />
      </form>
    </div>
  );
}

CourseFormComponent.propTypes = {
  authors: PropTypes.array,
  course: PropTypes.shape({
    title: PropTypes.string,
    id: PropTypes.number,
    slug: PropTypes.string,
    authorId: PropTypes.number
  })
};

export default CourseFormComponent;
