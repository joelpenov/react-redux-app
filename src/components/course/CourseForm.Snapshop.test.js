import React from "react";
import CourseFormComponent from "./CourseFormComponent";
import { authors, courses } from "../../../fakeApi/mockData";
import renderer from "react-test-renderer";

it("Should set button text to 'Saving...' when saving is true", () => {
  const tree = renderer.create(
    <CourseFormComponent
      authors={authors}
      courses={courses}
      course={courses[1]}
      saving
      handleOnSubmit={jest.fn()}
      onChange={jest.fn()}
      errors={{}}
    />
  );

  expect(tree).toMatchSnapshot();
});

it("Should set button text to 'Save' when saving is false", () => {
  const tree = renderer.create(
    <CourseFormComponent
      authors={authors}
      courses={courses}
      course={courses[1]}
      saving={false}
      handleOnSubmit={jest.fn()}
      onChange={jest.fn()}
      errors={{}}
    />
  );

  expect(tree).toMatchSnapshot();
});
