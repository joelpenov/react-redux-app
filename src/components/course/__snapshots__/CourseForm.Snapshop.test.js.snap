// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should set button text to 'Save' when saving is false 1`] = `
<div>
  <h1>
    Edit 
     course
  </h1>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="form-group"
    >
      <div
        className="col-md-6 mb-3"
      >
        <label
          htmlFor="title"
        >
          Course Name
        </label>
        <input
          className="form-control "
          id="title"
          name="title"
          onChange={[MockFunction]}
          placeholder="Course Name"
          type="text"
          value="React: The Big Picture"
        />
        
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="col-md-6 mb-3"
      >
        <label
          htmlFor="authorId"
        >
          Author
        </label>
        <select
          className="form-control "
          id="authorId"
          name="authorId"
          onChange={[MockFunction]}
          value={1}
        >
          <option
            value=""
          >
            -Please select one-
          </option>
          <option
            value={1}
          >
            Cory House
          </option>
          <option
            value={2}
          >
            Scott Allen
          </option>
          <option
            value={3}
          >
            Dan Wahlin
          </option>
        </select>
        
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="col-md-6 mb-3"
      >
        <label
          htmlFor="category"
        >
          Category
        </label>
        <input
          className="form-control "
          id="category"
          name="category"
          onChange={[MockFunction]}
          placeholder="Category"
          type="text"
          value="JavaScript"
        />
        
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Save
    </button>
  </form>
</div>
`;

exports[`Should set button text to 'Saving...' when saving is true 1`] = `
<div>
  <h1>
    Edit 
     course
  </h1>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="form-group"
    >
      <div
        className="col-md-6 mb-3"
      >
        <label
          htmlFor="title"
        >
          Course Name
        </label>
        <input
          className="form-control "
          id="title"
          name="title"
          onChange={[MockFunction]}
          placeholder="Course Name"
          type="text"
          value="React: The Big Picture"
        />
        
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="col-md-6 mb-3"
      >
        <label
          htmlFor="authorId"
        >
          Author
        </label>
        <select
          className="form-control "
          id="authorId"
          name="authorId"
          onChange={[MockFunction]}
          value={1}
        >
          <option
            value=""
          >
            -Please select one-
          </option>
          <option
            value={1}
          >
            Cory House
          </option>
          <option
            value={2}
          >
            Scott Allen
          </option>
          <option
            value={3}
          >
            Dan Wahlin
          </option>
        </select>
        
      </div>
    </div>
    <div
      className="form-group"
    >
      <div
        className="col-md-6 mb-3"
      >
        <label
          htmlFor="category"
        >
          Category
        </label>
        <input
          className="form-control "
          id="category"
          name="category"
          onChange={[MockFunction]}
          placeholder="Category"
          type="text"
          value="JavaScript"
        />
        
      </div>
    </div>
    <button
      className="btn btn-primary"
      disabled={true}
      type="submit"
    >
      Saving...
    </button>
  </form>
</div>
`;
